src/app/navigation/middleware.ts

```diff
@@ -31,7 +33,12 @@ export default (routes: AnyRoute[], history: History): Middleware => ({dispatch}
     history[action.payload.method]({
       hash: action.payload.hash,
       pathname: matchPathname(action.payload),
-      search: matchSearch(action.payload, action.payload.search),
+      search: matchSearch(
+        action.payload,
+        {
+          ...persistenQueryParameters(getState()),
+          ...queryString.parse(action.payload.search || ''),
```

Comment:
i mean this is less of a big deal but it'd be nice if we werent doing all this parsing and hashing of query arguments. all throughout the historyActions its a string but i'm not sure it has to be, i think it might be better as an object everywhere until the return result of `matchSearch`